clients/spark/core/src/test/scala/io/treeverse/gc/UncommittedGarbageCollectorSpec.scala

```diff
@@ -148,25 +148,63 @@ class UncommittedGarbageCollectorSpec
           UncommittedGarbageCollector.getFirstSlice(dataDF, repo) should be("")
         })
       }
+
+      it("should return empty first slice with only staged objects") {
+        withSparkSession(_ => {
+          val dataDir = new File(dir.toFile, "data")
+          dataDir.mkdir()
+          val legacySlice = repo + "_legacy_physical:address_path"
+          new File(dataDir, legacySlice).createNewFile()
+
+          val dataDF =
+            UncommittedGarbageCollector.listObjects(dir.toString,
+                                                    DateUtils.addHours(new Date(), +1)
+                                                   )
+          dataDF.count() should be(1)
+          UncommittedGarbageCollector.getFirstSlice(dataDF, repo) should be("")
+        })
+      }
+
+      it("should return empty first slice with only old repository data") {
+        withSparkSession(_ => {
+          val dataDir = new File(dir.toFile, "")
+          dataDir.mkdir()
+          val filename = "some_file"
+          new File(dataDir, filename).createNewFile()
+
+          val dataDF =
+            UncommittedGarbageCollector.listObjects(dir.toString,
+                                                    DateUtils.addHours(new Date(), +1)
+                                                   )
+          dataDF.count() should be(1)
+          UncommittedGarbageCollector.getFirstSlice(dataDF, repo) should be("")
+        })
+      }
+
       it("should return correct slice") {
         withSparkSession(_ => {
           val dataDir = new File(dir.toFile, "data")
           dataDir.mkdir()
           val legacySlice = repo + "_legacy_physical:address_path"
           val regularSlice = "xxx"
+          val regularSlice2 = "yyy"
           val filename = "some_file"
           var slice = new File(dataDir, legacySlice)
           slice.mkdir()
           new File(slice, filename).createNewFile()
           slice = new File(dataDir, regularSlice)
           slice.mkdir()
           new File(slice, filename).createNewFile()
+          slice = new File(dataDir, regularSlice2)
+          slice.mkdir()
+          new File(slice, filename).createNewFile()
 
-          val dataDF = UncommittedGarbageCollector.listObjects(dir.toString,
+          var dataDF = UncommittedGarbageCollector.listObjects(dir.toString,
```

Comment:
I think you should revert this change - dataDF should be a val and lines 206-7 can be written together as a single statement